components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/CacheBackedOrganizationManagementDAO.java

```diff
@@ -380,6 +380,13 @@ public int getRelativeDepthBetweenOrganizationsInSameBranch(String firstOrgId, S
         return organizationMgtDAO.getRelativeDepthBetweenOrganizationsInSameBranch(firstOrgId, secondOrgId);
     }
 
+    @Override
+    public String getParentOrganizationId(String organizationId, int depth)
```

Comment:
If this DAO method is accepting depth as a variable, rename the method to getAnAncestorOrganizationIdInGivenDepth
kind of thing

Reply:
If the method name is `getParentOrganizationId`, depth is not required.
write the query to return the parent id with depth = 1 for a given org id
If no records found return no parent

Reply:
Done